examples/simple_sptrsv.cpp

```diff
@@ -0,0 +1,58 @@
+#include <iostream>
+#include <spblas/spblas.hpp>
+
+#include <fmt/core.h>
+#include <fmt/ranges.h>
+
+int main(int argc, char** argv) {
+  using namespace spblas;
+
+  using T = float;
+
+  spblas::index_t m = 100;
+  spblas::index_t nnz_in = 20;
+
+  fmt::print("\n\t###########################################################"
+             "######################");
+  fmt::print("\n\t### Running SpTRSV Example:");
+  fmt::print("\n\t###");
+  fmt::print("\n\t###   solve for x:  A * x = alpha * b");
+  fmt::print("\n\t###");
+  fmt::print("\n\t### with ");
+  fmt::print("\n\t### A, in CSR format, of size ({}, {}) with nnz = {}", m, m,
+             nnz_in);
+  fmt::print("\n\t### x, a dense vector, of size ({}, {})", m, 1);
+  fmt::print("\n\t### b, a dense vector, of size ({}, {})", m, 1);
+  fmt::print("\n\t### using float and spblas::index_t (size = {} bytes)",
+             sizeof(spblas::index_t));
+  fmt::print("\n\t###########################################################"
+             "######################");
+  fmt::print("\n");
+
+  auto&& [values, rowptr, colind, shape, nnz] =
+      generate_csr<T, spblas::index_t>(m, m, nnz_in);
+
+  // scale values of matrix to make the implicit unit diagonal matrix
+  // be diagonally dominant, so it is solveable
+  T scale_factor = 1e-3f;
+  std::transform(values.begin(), values.end(), values.begin(), 
+                 [scale_factor](T val){ return scale_factor * val; });
```

Comment:
I think that with this scaling and using implicit_unit_diagonal_t, we should always be solveable.  We should consider adding a separate generator of some kind that guarantees it is lower and upper solveable without needing to go to such drastic changes ...